src/stacked_borrows/mod.rs

```diff
@@ -513,6 +513,14 @@ impl Stacks {
     }
 }
 
+impl VisitMachineValues for Stacks {
+    fn visit_machine_values(&self, visit: &mut ProvenanceVisitor) {
+        for tag in self.exposed_tags.iter().copied() {
```

Comment:
I'm a bit surprised by the 3 `copied()` you added, given that visiting should work by-ref.

Reply:
Refactoring churn.